extensions/npm/src/npmView.ts

```diff
@@ -241,7 +243,8 @@ export class NpmScriptsTreeDataProvider implements TreeDataProvider<TreeItem> {
 		if (!this.taskTree) {
 			const taskItems = await this.taskProvider.tasksWithLocation;
 			if (taskItems) {
-				this.taskTree = this.buildTaskTree(taskItems);
+				const taskTree = this.buildTaskTree(taskItems);
+				this.taskTree = this.sortTaskTree(taskTree);
```

Comment:
How about when we have just one item in the task tree. 
I think the sortTaskTree() will fail in that case.
Instead, with a condition in else, we can check if the length is 1 and ignore the sort function.

I have not tried running this code build. let me know if you tried it.  Will get back to it after I test the above-mentioned case.

Hope this is helpful.

Reply:
Hi @Ashok314,
It doesn't matter how many items you have in an array. I recommend you to check the specifications on how array sorting works with comparators.

https://tc39.es/ecma262/#sec-array.prototype.sort

https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm0CMAuGYCuBbARvATgNoC6MAvDCQNwCwAUKJLAtAExqa4EnmVLG0G4aHERQAzB2x4ipCoVYAaGP0H1GI0FgAOAQ3y6oIfLwAUAMwCW+aFK75lEeIwAmdmQEpyAPhgBvehggmHx4KAx8MBgrG1gAWhgnV0EAXzUhSBAAG3gAOiyQAHNTFigkXIhjKFMtPQMjfA8PQQ1svILi0tYKqpqQHX1DYyaW4Tb8opKxcR78atrBhpH6IA

Reply:
Thank you for your comment. 
I could have checked the specs before commenting. 

Thank you for the  TS playground code and specs source link.